extensions/ingestion/IngestionQueuePopulator.js

```diff
@@ -19,15 +21,42 @@ class IngestionQueuePopulator extends QueuePopulatorExtension {
             this.log.trace('skipping entry because missing bucket name');
             return;
         }
-        // Filter out bucket metadata entries
-        // If `attributes` key exists in metadata, this is a nested bucket
-        // metadata entry for s3c buckets
         if (entry.value) {
             const metadataVal = JSON.parse(entry.value);
+            // Filter out bucket metadata entries
+            // If `attributes` key exists in metadata, this is a nested bucket
+            // metadata entry for s3c buckets
             if (metadataVal.mdBucketModelVersion ||
                 metadataVal.attributes) {
                 return;
             }
+            // Filter out any master key object entries
+            if (entry.type === 'put') {
+                const queueEntry = new ObjectQueueEntry(entry.bucket,
+                                                        entry.key,
+                                                        metadataVal);
+                const sanityCheckRes = queueEntry.checkSanity();
+                if (sanityCheckRes) {
+                    this.log.trace('entry malformed', {
```

Comment:
Minor: the log message might also benefit by including the `sanityCheckRes.message`.